reactor-core/src/main/java/reactor/core/publisher/FluxRefCount.java

```diff
@@ -162,26 +165,33 @@ static final class RefCountInner<T>
 			implements QueueSubscription<T>, InnerOperator<T, T> {
 
 		final CoreSubscriber<? super T> actual;
-		final RefCountMonitor<T> connection;
 
+		RefCountMonitor<T> connection;
 		Subscription s;
 		QueueSubscription<T> qs;
 
-		volatile     int parentDone; //used to guard against doubly terminating subscribers (eg. double cancel)
-		static final AtomicIntegerFieldUpdater<RefCountInner> PARENT_DONE =
-				AtomicIntegerFieldUpdater.newUpdater(RefCountInner.class, "parentDone");
+		Throwable error;
+
+
+		static final int MONITOR_SET_FLAG = 0b0010_0000_0000_0000_0000_0000_0000_0000;
+		static final int TERMINATED_FLAG  = 0b0100_0000_0000_0000_0000_0000_0000_0000;
+		static final int CANCELLED_FLAG   = 0b1000_0000_0000_0000_0000_0000_0000_0000;
 
-		RefCountInner(CoreSubscriber<? super T> actual, RefCountMonitor<T> connection) {
+		volatile int state; //used to guard against doubly terminating subscribers (e.g. double cancel)
```

Comment:
I suppose the comment is no longer relevant, as the state represents more situations and before it was just about termination/cancellation, while now it also represents the readiness (monitor set). Perhaps we can remove it.

Reply:
fixed